package/src/frontend/media/playerSourceIDMap.js

```diff
@@ -0,0 +1,15 @@
+
+export const playerSourceIDMap = function (media) {  
+  return {
+
+    mapSources: function (id, sources) {
+      this[id] = sources;
+    },
+    get: function (sourceID, options) {
+      return media.getPlayer(this[sourceID], {
+        filePermaId: sourceID,
+        ...options
+      });
```

Comment:
`MultiPlayer` and this class will somehow have to collaborate to release players.

Reply:
As discussed I have introduced a mechanism in MultiPlayer pool i.e. playerSourceIDMap.
Now this pool maintains current and previous player. If there are two sources than pool 
keeps returning current and previous player. When third source is requested the first player is released and new one is requested.